### PR TITLE
Fix RESTful API NPE when restarting Kyuubi server

### PR DESCRIPTION
**Pull request by turboFei**
# :mag: Description
## Issue References 🔗

We meet issue when restarting the kyuubi server:
```
2024-08-28 09:48:41.507 WARN org.apache.kyuubi.server.api.RestExceptionMapper: Error occurs on accessing REST API.
java.lang.NullPointerException: Cannot invoke "org.apache.kyuubi.server.KyuubiServer.getConf()" because the return value of "org.apache.kyuubi.server.KyuubiServer$.kyuubiServer()" is null
	at org.apache.kyuubi.server.api.v1.BatchesResource.batchV2Enabled(BatchesResource.scala:72) ~[kyuubi-server_2.12-1.10.0.0.1.0.jar:1.10.0.0.1.0]
	at org.apache.kyuubi.server.api.v1.BatchesResource.$anonfun$batchInfo$3(BatchesResource.scala:345) ~[kyuubi-server_2.12-1.10.0.0.1.0.jar:1.10.0.0.1.0]
	at scala.Option.map(Option.scala:230) ~[scala-library-2.12.18.jar:?]
```

The root cause is that, the `KyuubiServer.kyuubiServer` is null until the kyuubi server start finished.

https://github.com/apache/kyuubi/blob/db57e9365d7933942197936ac7ff711d58f7ea91/kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiServer.scala#L231-L234

But the RESTful api initialization finished before than that, so, NPE will be thrown between REST front started to all the kyuubi service started.
## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

In this PR, 
1. move the KyuubiBatchService into KyuubiRestFrontendService
2. move the TempFileService (introduced in #6587) into KyuubiSessionManager

image

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by turboFei** [comment]
The UT passed.

gentle ping @pan3793 @bowenliang123

**Comment by turboFei** [comment]
also cc @wForget

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/TempFileService.scala [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.kyuubi.service
+package org.apache.kyuubi.server
```
Comment (bowenliang123) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/TempFileService.scala:18]: How about keeping the `TempFileService` class in the `org.apache.kyuubi.service` package?

**Review by bowenliang123 — APPROVED** [review]
LGTM.

**Comment by turboFei** [comment]
thanks, merging to master

**Comment by pan3793** [comment]
late lgtm too, thanks for fixing this.

**Pull request closed without merging**